### ISSUE TITLE
Navigation screen: can't move links into a submenu with keyboard only

### ISSUE THREAD

**Issue by tellthemachines** [WordPress/gutenberg#33271]
## Description

It should be possible to move a Navigation link into or out of a submenu using only the keyboard. The "Move to" block option, which exists for regular Navigation links, isn't showing on this screen:

<img width="222" alt="Screen Shot 2021-07-09 at 11 03 35 am" src="https://user-images.githubusercontent.com/8096000/125008916-dd7bc080-e0a6-11eb-9a5e-7ea1fe59da23.png">

## Step-by-step reproduction instructions

Add several items to Navigation, including submenus. Try to move items into or out of the submenus (or even from one submenu to another) using only the keyboard.

## Expected behaviour

Expect all block functionality to be available when using only keyboard.

## WordPress information
- WordPress version:  NA
- Gutenberg version:  11.0
- Are all plugins except Gutenberg deactivated?  Yes
- Are you using a default theme (e.g. Twenty Twenty-One)?  Yes

## Device information
- Device: 
- Operating system: 
- Browser:

**Comment by javierarce** [comment]
This is something that we should generalize for other blocks, like moving elements inside the column block, for example. 

In the particular case of the Navigation block, when a user clicks a link, the caret is automatically inserted, which means we cannot use the arrow keys to move the link around. So editing the text of the links should be a two-step process: with one click, you select the link (and can use the arrows to move it around), and with a second click, you edit the content (and the arrows let you move inside the text). 

Another option could be adding a modifier: <kbd>shift</kbd><kbd>→</kbd> or <kbd>shift</kbd><kbd>←</kbd> (same with up & down, depending on the direction of the navigation). With this solution I think users could move the links even when they are typing.

Here's a couple of related issues: 
- #17185
- #30899

**Comment by javierarce** [comment]
In a previous comment that I embarrassedly deleted from this issue (sorry!), I pointed out that moving elements (in general) should be extended to other blocks… I had forgotten about the "Keyboard shortcuts" list 🙈 and just tried the most obvious combinations (like <kbd>shift</kbd> <kbd>→</kbd>, <kbd>⌘</kbd> <kbd>←</kbd>, <kbd>⌘</kbd> <kbd>shift</kbd> <kbd>↑</kbd>, etc.) unsuccessfully. 

But it turns out, that the combination to move blocks up and down are <kbd>⇧</kbd> <kbd>⌥</kbd> <kbd>⌘</kbd> <kbd>T</kbd> and <kbd>⇧</kbd> <kbd>⌥</kbd> <kbd>⌘</kbd> <kbd>Y</kbd>. 

Leaving aside that those are complicated keyboard shortcuts (to type and to remember), I guess that to move things in and out submenus, we would need to come up with some shortcuts with a similar structure. Plus, if we could generalize them to be used elsewhere (to move blocks to their parent element, for instance), that would be a win-win.

**Comment by tellthemachines** [comment]
> In a previous comment that I embarrassedly deleted from this issue (sorry!)

No worries, just realised I messed up the issue description too 😂 

> But it turns out, that the combination to move blocks up and down are ⇧ ⌥ ⌘ T and ⇧ ⌥ ⌘ Y.

Those shortcuts only move blocks between their current siblings, not between nesting levels. But to move between nesting levels, we have Options > Move to which was implemented in #22453:

<img width="362" alt="Screen Shot 2021-07-09 at 11 02 58 am" src="https://user-images.githubusercontent.com/8096000/125008283-888b7a80-e0a5-11eb-8902-8c5ecd2423cf.png">

(Clicking the "Move to" button takes us into Select mode, where the left and right arrow keys can be used to move in and out of blocks and pressing Enter drops the block in the current cursor position.)

For some reason, I thought that option wasn't showing in the Navigation block children, but turns out it is! So this isn't an issue for the block, but it is for the screen, where that option doesn't show in the block options:

<img width="222" alt="Screen Shot 2021-07-09 at 11 03 35 am" src="https://user-images.githubusercontent.com/8096000/125008647-444caa00-e0a6-11eb-9971-136cfd7fd407.png">

I'll update this issue accordingly.

**Comment by Thelmachido** [comment]
Closing this issue due to the Navigation Screen project being moved to [an inactive status on the feature projects page](https://href.li/?https://make.wordpress.org/core/features/) in coordination with the project leads. If this work is picked back up, issues can be revisited and reopened as needed.
